CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -3,33 +3,22 @@
 #include <iostream>
 #include <pthread.h>
 #include <unistd.h>
+#include <vector>
 
 std::ifstream CallBack::_outFile;
 
 void* CallBack::OutputThread(void*)
 {
-  if (_outFile)
-  {
-    int pos = 0;
-    while (!_outFile.eof()) 
-    {
-      _outFile.seekg(0, _outFile.end);
-      int length = static_cast<int>(_outFile.tellg()) - pos;
-      _outFile.seekg(pos, _outFile.beg);
- 
-      if (length > 0) 
-      {
-        char* buffer = new char[length + 1];
-        _outFile.read(buffer, length);
-        buffer[length] = '\0';
-        pos += length;
-        std::cout << buffer;
-        delete[] buffer;
-      }
+  if (_outFile.is_open()) {
+    std::string line;
+    while (true) {
+        while (std::getline(_outFile, line)) std::cout << line << std::endl;
+        if (!_outFile.eof()) break; // Ensure end of read was EOF.
+        _outFile.clear();
```

Comment:
@JoshDreamland Take a look over this it's working great now and Travis and AppVeyor have both stopped segfaulting. They also no longer print in big blocks, but print more regularly now. In addition to working better it's also a lot shorter and more coherent.